core/src/banking_stage.rs

```diff
@@ -2522,7 +2522,11 @@ mod tests {
                 let packet_batches = convert_from_old_verified(packet_batches);
                 verified_sender.send(packet_batches).unwrap();
 
-                sleep(Duration::from_millis(200));
+                // wait for banking_stage to eat the packets
+                while bank.get_balance(&alice.pubkey()) < 1 {
```

Comment:
```suggestion
                while bank.get_balance(&alice.pubkey()) < 2 {
```
This should be 2 and below should be 3

Reply:
I think it's correct currently in the PR. The first transaction will bring the balance to 1. So it'll loop as long as the balance is < 1.

The second transaction will bring up the balance to 2. So the second loop waits till the balance reaches 2.

Reply:
Really? The first transaction transfers 2 lamports and the second transaction transfers 1

Reply:
Hmm, ok. Let me take another look. Currently viewing it from my phone.

Reply:
Yes, you are right. I missed the transaction details. I'll update the test and push an update.

Thanks for catching it.

Reply:
No problem, I think you can decrease the sleep duration as well btw

Reply:
After reading the test more, I have decided to revert it to the original flow (prior to banking stage change).
The test is ensuring that only one of the two transactions is successful. But, the test checks/conditions have to be updated. 

Earlier it was expecting that the first transaction is successful and the second fails (since the banking stage was reading and processing one transaction batch at a time). But with greedy receive, both the batches are received in a single iteration, and any of the two transactions could be processed first.

So, flow of the test is reverted to the original flow. The checks for balance has been updated to ensure that only one of the transaction is processed.